### PR TITLE
Add d2l-ajax-framed

### PR DESCRIPTION
**Pull request by ryantmer**
Uses frau-jwt to be able to work in iframes. This allowed removing a bunch of code, since frau-jwt takes care of XSRF tokens and caching itself.

This was added as a separate components for two reasons - first, as there doesn't seem to be a reliable way of determining if we are in an iframe, especially with WCs (the usual checks aren't reliable), and secondly because it greatly increases the size, due to the browserifying of frau-jwt and its dependencies.

In theory, we could probably use frau-jwt/framed for d2l-ajax-framed, and frau-jwt/local for the normal d2l-ajax, which wouldn't be overly large.

Things that could be improved:

- use `frau-jwt/framed` specifically, since this is (presumably) always going to be used in an iframe
- ~~make sure it works, because... yeah I don't know that it works.~~ Confirmed working by @RyanSweetD2L and @lawong-d2l

TODO:

- [x] Use `frau-jwt/framed` to avoid packing `superagent` (which is only used by `frau-jwt/local`) - see https://github.com/Brightspace/frau-jwt/pull/13
- [x] Add tests
- [ ] Get `ugligyify` working
- [ ] Follow-up PR: use `frau-jwt/local` for the regular `d2l-ajax`, as it does a lot of what we do here.

### CODE REVIEW

**Comment by ryantmer** [comment]
Upon a closer look, tests here would be like... 99% just testing `frau-jwt`, so that's rather pointless. Now just waiting on https://github.com/Brightspace/frau-jwt/pull/13 for this, then should be able to release it.

File: package.json [hunk 1]
```diff
@@ -12,12 +13,16 @@
   },
   "author": "D2L Corporation",
   "license": "Apache-2.0",
-  "dependencies": {},
+  "dependencies": {
+    "frau-jwt": "^1.1.0"
```
Comment (omsmith) [inline package.json]: Can you 1.1.1 now :fast_forward:

File: d2l-ajax-framed.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	},
+	generateRequest: function() {
+		jwt(this.scope)
+			.then(function(token) {
+				this.authToken = token;
+				this.$.request.generateRequest();
+			}.bind(this))
+			.catch(function(e) {
+				this.onError(e);
+			});
```
Comment (omsmith) [inline d2l-ajax-framed.js]: This needs a `.bind` as well, no?

File: d2l-ajax-framed.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+		if (headers instanceof Object) {
+			for (header in headers) {
+				result[header] = headers[header].toString();
+			}
+		}
+
+		return result;
+	},
+	generateRequest: function() {
```
Comment (omsmith) [inline d2l-ajax-framed.js]: Is there a way for me to `generateRequest` twice on this, such that `this.$.request.generateRequest()` hasn't fired off and I'm modifying other bits and bobs already?

File: d2l-ajax-framed.js (hunk 2 above)
Comment (ryantmer) [inline d2l-ajax-framed.js]: Sure does! Yep.

**Pull request merged**